### DISCUSSION TITLE
Introduction / Willing to participate / Asking for help beyond the documentation

### DISCUSSION THREAD

**Discussion by ThierryMGTX** [horizon-eda/horizon#794] [General]
Hi @carrotIndustries (and everybody).

Willing to learn (and already starting) your EDA software tool for hobby use, I'd like to introduce myself and ask for some questions/help, see below.
First, I'm a bit confused by the structure of the discussions within GitHub and (here) a lot of help requests seem to be posted as issues ; therefore I'm asking for advices of where to properly post my questions. Feel free to instruct me to rewrite my post and to split and spread it to the appropriate places.

**Now my introduction :**
In electronics design (hobby & pro), old skills in programming (PC and low-level like embedded), using GNU/LINUX at a basic/intermediate level.
Currently using Altium Designer at my work (with an external database for working around some of the bad library structuring), I'm motivated in learning the use of an open-source EDA software tool in order to be able to design clean electronic circuits outside of my work, with some requirements like a "decent" ergonomy (selecting objects, editing properties etc... where KiCad fails), hierarchical schematics... and _(what seduced me in the introduction Horizon EDA) a correct redundancy-free library structure_, somehow in the spirit of a relational database (KiCad fails too ; only some proprietary tools like EAGLE, Pulsonix and its derivatives succeeded), not counting, of course, push-and-shove interactive routing (which is now a standard requirement even for non-commercial tools).

Therefore, after installing the software (staying at v2.4 from an official repository because containerized install adds too much complexity) and making it work properly (compact GTK theme required)...
... I'm also slowly trying to re-learn every basic EDA software basic task, beginning with creating new library parts and all of the associated object types : despite the existence of the official pool, we always need to create more new parts as we start using the software for the first times, and, according to my own experience (pro. included), we absolutely need to build a rich connector library (and that's why I'd like to contribute to the official pool for that, whereas I found difficulties, not counting that I don't currently master GIT in particular).
However, although I'm aware I have to re-learn the ways to achieve every task (which differ if with respect of in the software tools I already know), I knock my head in each and every obstacle during this learning phase, despite reading the official documentation and searching the Internet (where, unfortunately, the Horizon-EDA community is not yet big enough), therefore needing for help about targeted but not yet covered topics. A few examples :

- including 3D models in footprints (as used in Altium Designer !), I can't figure out the workings of the 3D management interface elements at a first glance, and/or some basic operating procedures / workflows, because it is not yet covered by the documentation
- there is an interesting "3D projection" feature, as seen in the blog, but I can't use it because the checkbox for it is permanently disabled. By the way, how does this feature work ? Is it only a visual help ? Which adds snapping points for 2D drawing, or not ? Which is able to generate graphic primitives for pasting into the package outline layer ?

I would also have to ask a lot of further questions form time to time, and possibly post enhancement ideas as well (hoping they could fit into the software architecture (I'm rather aware of such concerns, having tried myself to design software, however not as complex as this EDA tool... That's why I really praise the colossal work and ahead of time thinking already done by @carrotIndustries !)

Best Regards.

Thierry

**Comment by carrotIndustries** [comment]
> including 3D models in footprints (as used in Altium Designer !), I can't figure out the workings of the 3D management interface elements at a first glance, and/or some basic operating procedures / workflows, because it is not yet covered by the documentation

That's the first time I'm hearing this. Which parts couldn't you figure out?

> can't use it because the checkbox for it is permanently disabled.

You first have to click "Project" in the 3D view. The button is next to the model.

**Reply by ThierryMGTX** [reply]
Hi Lukas.

Sorry for my late answer : although it seems possible to directly reply by E-Mail, it doesn't seem to work, so that I'll retry directly here on the website :

> > including 3D models in footprints (as used in Altium Designer !), I can't figure out the workings of the 3D management interface elements at a first glance, and/or some basic operating procedures / workflows, because it is not yet covered by the documentation
> 
> That's the first time I'm hearing this. Which parts couldn't you figure out?

Oops ! Indeed, it's not so complicated (I was perhaps tired and had lost patience after hitting many obstacles on my previous "session" on the software) : I managed to adjust the Z coordinate ([from], picking a vertex that should touch the board surface, then [to], checking Z=0, and at last, doing the move by pressing the button below) and I now get a foot on the ladder in order to continue...
I think I understand now : the [from] and [to] define a translation vector (or its various projections when disabling one or many [to] coordinates) you apply on the 3D model : a simple but polyvalent tool !
By the way, "pick two (center)" is a well-thought-out feature, according to my experience in A.D., in which I repeatedly do something similar, usually for finding (and using) the center of the cross-section of a pin. 
 
> > can't use it because the checkbox for it is permanently disabled.
> 
> You first have to click "Project" in the 3D view. The button is next to the model.

What is the meaning behind this button ? A library package seems self-contained, without any relationship to a project.

Further question : the 2D projection is only a visual help, isn't it (there seems to be no associated 2D layer, although we can snap on it) ?
What about the ability to generate a raw package outline from it (this would copy-paste all primitives, so that the user can edit/delete some of them afterwards, according to the desired level of details) ?

Other question (I need advices in order to get started because I'm not used to online collaboration for development) : should I better open "issues", one per separate question, as it seems to be the case for other users ? There also seems to be a "label" system for telling if an "issue" is a bug, enhancement request, question etc... but nobody seems to use it (any attempt to filter by label results in empty search results). 

Again, thanks in advance... and happy New Year...

**Reply by carrotIndustries** [reply]
> What is the meaning behind this button ? A library package seems self-contained, without any relationship to a project.-

Project is a verb (create projection), not a noun in this instance.

The geometry resulting from the projection isn't really suitable for package outlines as-is and often needs to be simplified, that's why it can only be used for snapping.

**Reply by ThierryMGTX** [reply]
Ah, the trap of homographs ! Then, in order to disambiguate the English translation, I would have no other choice to use a longer caption like "projection", "2D proj." or a tiny icon representing e.g. a cube with dotted lines to a square.

2D projection : you're right indeed, especially in the case of a detailed 3D model which would generate too many tiny primitives.
But in the case of a simplified 3D model (as I encountered), I would reuse almost all of the details (I like to draw more detailed outlines or silkscreen, especially for connectors, in order to show keyways or other features showing the orientation).

And what about my last question ? (should I use the issues instead ?)

Thanks...
